### PR TITLE
Turn Workload Type and Service Type value sets into enums

### LINKED ISSUE #532: Turn Workload Type and Service Type value sets into enums
There are a few cases like the following: https://github.com/appvia/kev/blob/5a1c05c568427024fb3ef76a9fc0a5d6bdd1e341/pkg/kev/config/extensions.go#L187

Where the return value is a string and the possible values can only be obtained by reading the code and following all the different options. 

This could be transformed into Enums and supported by a few additional functions which ensure that conversion using different cases works (eg None, noNe, etc) and that there is a Valid() function that can quickly assert validity of a specific string value.

This ticket deals with Restart Policy.

- [x] 1. Workload Type

- [x] 2. Service Type 

Should review the whole configuration tree

### PR DESCRIPTION
**Pull request by ezodude**
Resolves #532 

- Workload Types are now enums, and can handle case insensitive k8s extension config.
- Service Types are now enums, and can handle case insensitive k8s extension config.

### CODE REVIEW

File: pkg/kev/config/svcext.go [hunk 1]
```diff
@@ -260,18 +267,18 @@ func PodSecurityWithDefaults() PodSecurity {
 	}
 }
 
-func ServiceTypeFromCompose(svc *composego.ServiceConfig) (string, error) {
-	serviceType := NoService
+func ServiceTypeFromCompose(svc *composego.ServiceConfig) (ServiceType, error) {
+	var candidate = "none"
```
Comment (marcinc) [inline pkg/kev/config/svcext.go:271]: What's the reason for this change (and the other cases below)?

File: pkg/kev/converter/kubernetes/project_service.go [hunk 2]
```diff
@@ -85,11 +85,11 @@ func (p *ProjectService) workloadType() string {
 }
 
 // serviceType returns service type for project service workload
-func (p *ProjectService) serviceType() (string, error) {
+func (p *ProjectService) serviceType() (config.ServiceType, error) {
 	serviceType := p.SvcK8sConfig.Service.Type
 
 	// @step validate whether service type is set properly when node port is specified
-	if !strings.EqualFold(serviceType, string(v1.ServiceTypeNodePort)) && p.nodePort() != 0 {
+	if !config.ServiceTypesEqual(serviceType, config.NodePortService) && p.nodePort() != 0 {
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/project_service.go:92]: minor comment, `v1.ServiceTypeNodePort` is unlikely to change in contrast to the value of `config.NodePortService`... I know they both represent the same thing but perhaps we should stick to the former?

**Review by marcinc — COMMENTED** [review]
couple of tiny comments.

File: pkg/kev/config/svcext.go (hunk 1 above)
Comment (ezodude) [reply pkg/kev/config/svcext.go:271]: I've reverted to using the raw values determined by the various rules.

Then, `inferServiceTypeFromComposeValue` can use its own internal rules to turn the candidate raw value into the actual `ServiceType` enum.

Otherwise, it can get confusing as to why I'm using an enum to start with. Changing it to a string, and then `inferServiceTypeFromComposeValue` changing it back to an enum.

**Review by marcinc — APPROVED** [review]
